service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/DraftOrdersControllerDateOfIssueMidEventTest.java

```diff
@@ -0,0 +1,208 @@
+import static uk.gov.hmcts.reform.fpl.enums.DirectionAssignee.PARENTS_AND_RESPONDENTS;
+import static uk.gov.hmcts.reform.fpl.enums.JudgeOrMagistrateTitle.HIS_HONOUR_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(DraftOrdersController.class)
+@OverrideAutoConfiguration(enabled = true)
+class DraftOrdersControllerDateOfIssueMidEventTest extends AbstractControllerTest {
+
+    DraftOrdersControllerDateOfIssueMidEventTest() {
+        super("draft-standard-directions");
+    }
+
+    @Nested
+    class DateOfIssueValidation {
+
+        @Test
+        void shouldReturnErrorsWhenTheDateOfIssueIsInFuture() {
+            CaseDetails caseDetails = caseDetails(now().plusDays(1));
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).containsOnlyOnce("Date of issue cannot be in the future");
+        }
+
+        @Test
+        void shouldNotReturnErrorsWhenDateOfIssueIsToday() {
+            CaseDetails caseDetails = caseDetails(now());
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).isNull();
+        }
+
+        @Test
+        void shouldNotReturnErrorsWhenDateOfIssueIsInPast() {
+            CaseDetails caseDetails = caseDetails(now().minusDays(1));
+            AboutToStartOrSubmitCallbackResponse response = postMidEvent(caseDetails, "date-of-issue");
+            assertThat(response.getErrors()).isNull();
+        }
+
+        private CaseDetails caseDetails(LocalDateTime localDateTime) {
+            return CaseDetails.builder().data(Map.of("dateOfIssue", localDateTime.toLocalDate())).build();
+        }
+    }
+
+    @Test
+    void shouldPopulateCorrectHearingDate() {
+        LocalDateTime hearingDate = LocalDateTime.of(2020, 1, 1, 0, 0, 0);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(Map.of("hearingDetails", wrapElements(createHearingBooking(hearingDate, hearingDate.plusDays(1)))))
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+
+        Stream.of(DirectionAssignee.values()).forEach(assignee ->
+            AssertionsForClassTypes.assertThat(callbackResponse.getData().get(assignee.toHearingDateField()))
+                .isEqualTo("1 January 2020, 12:00am"));
+    }
+
+    @Test
+    void shouldShowEmptyPlaceHolderForHearingDateWhenNoHearingDate() {
+        CaseDetails caseDetails = CaseDetails.builder().data(new HashMap<>()).build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = postMidEvent(caseDetails, "date-of-issue");
+
+        Stream.of(DirectionAssignee.values()).forEach(assignee ->
```

Comment:
Seeing as this stream is used more than one, thoughts of precomputing the resultant array of values at the start of the class creation.
```java
    private static final Object[] DIRECTION_ASSIGNEE_KEYS = Stream.of(DirectionAssignee.values())
        .map(DirectionAssignee::toHearingDateField)
        .toArray();
```
and then the assertion can be
```java
assertThat(callbackResponse.getData()).extracting(DIRECTION_ASSIGNEE_KEYS).containsOnly("Please enter a hearing date");
```
same can then be used for the test above

Reply:
meh